python/kfserving/kfserving/storage.py

```diff
@@ -68,6 +68,7 @@ def _download_s3(uri, temp_dir: str):
         bucket_name = bucket_args[0]
         bucket_path = bucket_args[1] if len(bucket_args) > 1 else ""
         objects = client.list_objects(bucket_name, prefix=bucket_path, recursive=True)
+        count = 0
```

Comment:
Alteranatively:

if len(objects) == 0: 
  raise..

Reply:
Seems the returned objects from s3/blob/gcs are either iterator or generator, so we can not use len to them and here to make it consistent use the same method to check whether it really returned something.

Reply:
SGTM